spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -94,10 +95,14 @@
   private boolean useStartingSequenceNumber;
   private RewriteJobOrder rewriteJobOrder;
   private RewriteStrategy strategy = null;
+  private final SparkSession cloneSession;
 
   RewriteDataFilesSparkAction(SparkSession spark, Table table) {
     super(spark);
```

Comment:
I think this needs to be the cloned session, otherwise the subclass calls would be modifying the original session and not the new cloned one correct?

Reply:
I think not using the same session will probably also have some impacts on some of the other BaseSparkAction functions like "withJobGroupInfo"

Reply:
I agree. In general, I think this should probably be overriding `spark()` so that the same context is used everywhere consistently.